### PR TITLE
Feature/roles

### PR DESCRIPTION
**Pull request by dereklwood**
- [x] porting the roles endpoint (all but `get_metrics`), plus a few supporting types
- [ ] the python version has no direct tests for this endpoint

### CODE REVIEW

File: ruby/lib/endpoints/roles.rb [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+          'roleConfigGroupRef' => ROAttr.new(ApiRoleConfigGroupRef),
+          'zookeeperServerMode' => ROAttr.new,
+          'entityStatus' => ROAttr.new
+        }
+
+        def initialize(resource_root, name = nil, type = nil, hostRef = nil)
+          super(resource_root, { name: name, type: type, hostRef: hostRef })
+        end
+
+        def to_s
```
Comment (wolf31o2) [inline ruby/lib/endpoints/roles.rb:115]: to_str

File: ruby/lib/endpoints/types.rb [hunk 2]
```diff
@@ -576,6 +576,36 @@ def abort
...
+        @_ATTRIBUTES = {
+          'name' => ROAttr.new,
+          'argSchema' => ROAttr.new
+        }
+
+        def initialize(resource_root)
+          super(resource_root)
+        end
+
+        def to_s
```
Comment (wolf31o2) [inline ruby/lib/endpoints/types.rb:606]: to_str

**Comment by wolf31o2** [comment]
Minor comments. Otherwise, 👍 LGTM

File: ruby/lib/endpoints/roles.rb (hunk 1 above)
Comment (dereklwood) [inline ruby/lib/endpoints/roles.rb:115]: `to_s` is more appropriate here, as in https://github.com/caskdata/cm_api/pull/1#discussion_r68842682.  All these ApiObjects will provide these string representations used in logging, but they shouldn't need to have any string operations performed on them

**Pull request merged**